purchase_allowed_product/models/product_supplierinfo.py

```diff
@@ -7,4 +7,4 @@
 class ProductSupplierinfo(models.Model):
     _inherit = "product.supplierinfo"
 
-    name = fields.Many2one("res.partner", index=True)
+    partner_id = fields.Many2one("res.partner", index=True)
```

Comment:
No need of the comodel name and put a comment to know what is happening.

```suggestion
    # Index this field
    partner_id = fields.Many2one(index=True)
```

Reply:
done! thank you @pedrobaeza